components/script/dom/htmlscriptelement.rs

```diff
@@ -148,9 +231,22 @@ pub enum ScriptType {
     Module,
 }
 
+#[derive(JSTraceable, MallocSizeOf)]
+pub struct CompiledSourceCode {
```

Comment:
We either need to add a `#[must_root]` to this struct (and store it within a `RootedTraceableBox`) since it contains a GC value, or we need to store `pub source_code: RootedTraceableBox<Heap<*mut JSScript>`.

Reply:
I think we are only passing this around, so then I guess using the `RootedTraceableBox` is the right option? I thougth a `Heap` would also root the value, however I guess that is not correct, and most use of `Heap` is within a dom struct which roots it...

Reply:
Storing the RootedTraceableBox as a member of CompiledSourceCode should be the easiest solution.